pallets/parachain-staking/src/lib.rs

```diff
@@ -1803,6 +1831,21 @@ pub mod pallet {
 			Ok(())
 		}
 
+		fn delegator_is_kicked(delegation: StakeOf<T>, collator: &T::AccountId) -> DispatchResult {
```

Comment:
I think `delegation` can be a `&StakeOf<T>`, so that we avoid cloning when we want to use it.